app/components/ui/button.tsx

```diff
@@ -0,0 +1,63 @@
+import type { ButtonHTMLAttributes } from 'react';
+
+type ButtonProps = ButtonHTMLAttributes<HTMLButtonElement> & {
+  variant: 'primary' | 'secondary' | 'warm' | 'outlined';
+  buttonStyle: 'rounded' | 'fullyRounded';
+  size: 'small' | 'medium' | 'large';
+  children: React.ReactNode;
+};
+
+export function Button({ variant, buttonStyle, size, children, ...props }: ButtonProps) {
+  const baseClasses = 'w-full sm:w-fit me-2 mb-2 text-center font-medium focus:outline-none focus:ring-4';
+  let variantClasses = '';
+  let styleClasses = '';
+  let sizeClasses = '';
+
+  switch (variant) {
```

Comment:
Just a nit pick and not a blocker: I think `if-else` statements would work a bit better here since we don't really have a `default` case (as `variant` is a required property). Or, as @eiffelwong1 suggests, a map!

Reply:
(Same with the other switch statements)

Reply:
Agree with the suggestions! First, I did it with if..else statements, then switched as cases seemed more readable. Gonna back to them. 😄 

I read we can use case without default when you 100% sure about your cases and you set default elements in TypeScript as a fallback. Haven't heard about the map thing though. Good to know!